File: assignments/python/grade-school/grade_school_test.py
try:
    from school import School
except ImportError:
    raise SystemExit('Could not find school.py. Does it exist?')

import unittest

class SchoolTest(unittest.TestCase):
    def setUp(self):
        self.school = School("Haleakala Hippy School")
  
    def test_an_empty_school(self):
        self.assertEqual({}, self.school.db)
  
    def test_add_student(self):
        self.school.add("Aimee", 2)
        self.assertEqual({2: {"Aimee"}}, self.school.db)
  
    def test_add_more_students_in_same_class(self):
        self.school.add("James", 2)
        self.school.add("Blair", 2)
        self.school.add("Paul", 2)
        self.assertEqual({2: {"James", "Blair", "Paul"}}, self.school.db)
  
    def test_add_students_to_different_grades(self):
        self.school.add("Chelsea", 3)
        self.school.add("Logan", 7)
        self.assertEqual({3: {"Chelsea"}, 7: {"Logan"}}, self.school.db)
  
    def test_get_students_in_a_grade(self):
        self.school.add("Franklin", 5)
        self.school.add("Bradley", 5)
        self.school.add("Jeff", 1)
        self.assertEqual({"Franklin", "Bradley"}, self.school.grade(5))
  
    def test_get_students_in_a_non_existant_grade(self):
        self.assertEqual(set(), self.school.grade(1))
  
    def test_sort_school(self):
        self.school.add("Jennifer", 4)
        self.school.add("Kareem", 6)
        self.school.add("Christopher", 4)
        self.school.add("Kyle", 3)
        sorted_students = {
          3: ("Kyle",),
          4: ("Christopher", "Jennifer",),
          6: ("Kareem",)
        }
        self.assertEqual(sorted_students, self.school.sort())

if __name__ == '__main__':
    unittest.main()

File: assignments/ruby/grade-school/grade_school_test.rb
require 'minitest/autorun'
require_relative 'school'

class SchoolTest < MiniTest::Unit::TestCase

  def school
    @school
  end

  def setup
    @school = School.new
  end

  def test_an_empty_school
    assert_equal({}, school.db)
  end

  def test_add_student
    skip
    school.add("Aimee", 2)
    assert_equal({2 => ["Aimee"]}, school.db)
  end

  def test_add_more_students_in_same_class
    skip
    school.add("James", 2)
    school.add("Blair", 2)
    school.add("Paul", 2)
    assert_equal({2 => ["James", "Blair", "Paul"]}, school.db)
  end

  def test_add_students_to_different_grades
    skip
    school.add("Chelsea", 3)
    school.add("Logan", 7)
    assert_equal({3 => ["Chelsea"], 7 => ["Logan"]}, school.db)
  end

  def test_get_students_in_a_grade
    skip
    school.add("Franklin", 5)
    school.add("Bradley", 5)
    school.add("Jeff", 1)
    assert_equal ["Franklin", "Bradley"], school.grade(5)
  end

  def test_get_students_in_a_non_existant_grade
    skip
    assert_equal [], school.grade(1)
  end

  def test_sort_school
    skip
    school.add("Jennifer", 4)
    school.add("Kareem", 6)
    school.add("Christopher", 4)
    school.add("Kyle", 3)
    sorted = {
      3 => ["Kyle"],
      4 => ["Christopher", "Jennifer"],
      6 => ["Kareem"]
    }
    assert_equal sorted, school.sort
    assert_equal [3, 4, 6], school.sort.keys
  end
end
